batcher-ui/src/pages/Main.tsx

```diff
@@ -82,23 +84,98 @@ const Welcome: React.FC = () => {
         if (status === BatcherStatus.OPEN) {
           setOpenTime(jsonData.value.status.open);
         }
-
-        console.log('%cMain.tsx line:114 jsonData.value', 'color: #007acc;', jsonData.value);
         setOrderBook(jsonData.value.orderbook);
       }
     } catch (error) {
       console.log('Batcher error', error);
     }
   };
 
+  const getClearedPayoutOfUserAddress = async () => {
+    if (!userAddress) {
+      setSellSideAmount(0);
+      setBuySideAmount(0);
+      return;
+    }
+
+    const storage = await contractsService.getStorage({
+      address: contractAddress,
+      level: 0,
+      path: null,
+    });
+
+    const userOrderBookURI = bigMapsByIdUri + storage.user_orderbook + '/keys/' + userAddress;
+    const userOrderBookData = await fetch(userOrderBookURI, { method: 'GET' });
+    const userOrderBooks = await userOrderBookData.json();
+    if (!Array.isArray(userOrderBooks.value.open) || userOrderBooks.value.open.length == 0) {
+      return;
+    }
+
+    const openOrderBooks = userOrderBooks.value.open;
+
+    console.log(344, openOrderBooks);
+
+    let initialBuySideAmount = 0;
+    let initialSellSideAmount = 0;
+
+    for (var i = 0; i < openOrderBooks.length; i++) {
+      const batchURI =
+        bigMapsByIdUri + storage.batch_set.batches + '/keys/' + openOrderBooks.at(i).batch_number;
+      const batchData = await fetch(batchURI, { method: 'GET' });
+      const batch = await batchData.json();
+      console.log(1444, batch);
+
+      const clearingRate =
+        parseInt(batch.value.status.cleared.rate.rate.val) *
+        10 ** parseInt(batch.value.status.cleared.rate.rate.pow);
+
+      const clearingKey = Object.keys(batch.value.status.cleared.clearing.clearing_tolerance)[0];
```

Comment:
We need to check if their order was in those levels that were interested in the clearing price.  If they are outside of the clearing price then they don't get filled and we can just transfer their original amount without doing any further calculation.  Logic for this is in the contract treasury

Reply:
I get the `batches` that this current user includes in the open section and get the exciting rate of this batch.